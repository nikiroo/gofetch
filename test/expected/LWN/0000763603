      [$] MEASURING (AND FIXING) I/O-CONTROLLER THROUGHPUT LOSS    

  [Kernel] Aug 29, 2018 21:20 UTC (Wed) (corbet)

  o Reference: 0000763603
  o News link: https://lwn.net/Articles/763603
  o Source link: 


    [$]   Sorry,   this  article  is  currently  available  to  LWN
    suscribers only [https://lwn.net/subscribe/].



