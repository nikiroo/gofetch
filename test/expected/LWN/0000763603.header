0[$] Measuring (and fixing) I/O-controller throughput loss	null/LWN/0000763603		70
i  [Kernel] Aug 29, 2018 21:20 UTC (Wed) (corbet)
i
i    Many  services,  from  web hosting and video streaming to cloud
i    storage,  need  to  move  data  to  and from storage. They also
i    often  require  that  each  per-client I/O flow be guaranteed a
i    non-zero   amount  of  bandwidth  and  a  bounded  latency.  An
i    expensive  way to provide these guarantees is to over-provision
i    storage  resources,  keeping  each  resource underutilized, and
i    thus  have  plenty of bandwidth available for the few I/O flows
i    dispatched  to  each  medium.  Alternatively one can use an I/O
i    controller.  Linux provides two mechanisms designed to throttle
i    some  I/O  streams  to allow others to meet their bandwidth and
i    latency  requirements.  These mechanisms work, but they come at
i    a  cost:  a  loss  of  as  much  as  80% of total available I/O
i    bandwidth.  I  have run some tests to demonstrate this problem;
i    some  upcoming improvements to the bfq I/O scheduler promise to
i    improve the situation considerably.
i
