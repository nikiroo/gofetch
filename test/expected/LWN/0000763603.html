<!DOCTYPE html>
<html>
<head>
  <meta http-equiv='content-type' content='text/html; charset=utf-8'>
  <meta name='viewport' content='width=device-width, initial-scale=1.0'>
  <style type='text/css'>
    body { margin: 1em 15%; }
  </style>
</head>
<body>
<div class='story'>
	<h1>[$] Measuring (and fixing) I/O-controller throughput loss</h1>
	<div class='details'>([Kernel] Aug 29, 2018 21:20 UTC (Wed) (corbet))</div>
	<br/>
    <ul>
        <li>News link: <a href='https://lwn.net/Articles/763603'>https://lwn.net/Articles/763603</a></li>
        <li>Source link: <a href=''></a></li>
    </ul>
	<br/>
	<div class='content' style='text-align: justify'>
		[$] Sorry, this article is currently available to LWN suscribers only [https://lwn.net/subscribe/].
	</div>
<hr/>
</div>
</body>
